### PR TITLE
Add config option to show highlights only when sidebar is open

### PR DESCRIPTION
**Pull request by robertknight**
This PR implements a configuration option to only show highlights automatically when the sidebar is open. When the sidebar is open or closed the user can still toggle highlights on or off via the sidebar's vertical toolbar. The purpose of this is to enable a cleaner reading experience by default until the user chooses to view the annotation layer.

The "showHighlights" boolean configuration option has been changed to alternatively accept the enum values "always", "never" or "whenSidebarOpen". The boolean values `true` and `false` are mapped to `"always"` and `"never"` respectively.

The intended use of this option is that the sidebar is configured such that it will not overlap the main content of the page at the initial width. No such option currently exists, that will be a separate task.

Fixes https://github.com/hypothesis/product-backlog/issues/273

### CODE REVIEW

**Comment by seanh** [comment]
The _Show Highlights_ button seems to be overridden by the sidebar opening and closing:

* With the sidebar closed: click the _Show Highlights_ button to turn highlights on, open the sidebar, close the sidebar again: highlights get turned off.

* With the sidebar closed: click the _Show Highlights_ button to turn highlights _off_, open the sidebar: highlights get turned on.

But from the acceptance criteria:

> eye button will override initial site preference. Turning highlights on via the button will keep highlights visible until the user uses the button again to turn highlights and vice versa (for this page load, subsequent loads will take initial site preference)

In other words as soon as the user clicks the _Show Highlights_ button once the whole automatically turning highlights on and off when the sidebar opens and closes behaviour should get disabled.

(But note that the preference does _not_ need to be persisted across page loads.)

**Comment by seanh** [comment]
Code-wise looks great

**Comment by seanh** [comment]
As [discussed on Slack](https://hypothes-is.slack.com/archives/C1M8NH76X/p1495531467244934) mergin this as it'll do for what we need it for
